pkg/controller/signup.go

```diff
@@ -60,3 +62,38 @@ func (s *Signup) GetHandler(ctx *gin.Context) {
 		ctx.JSON(http.StatusOK, signupResource)
 	}
 }
+
+// VerifyCodeHandler validates the phone verification code passed in by the user
+func (s *Signup) VerifyCodeHandler(ctx *gin.Context) {
+	code := ctx.Param("code")
+
+	userID := ctx.GetString(context.SubKey)
```

Comment:
could we also add "code" to the context used here by any chance?

Reply:
i thought at the time of creating context the idea was so that we wouldn't be hard-coding strings when trying to retrieve values from the context, an example is FamilyNameKey = "familyName" for instance..

Reply:
Putting params to the context is a way to pass those params trough the code.
Every secured endpoint requires a user token. So, it makes sense to extract all the information from the token (id, username, email, name, etc) once and put it to the context, so, the endpoints do not have to parse the token individually.
"code" is specific to this verify endpoint. It won't be available in the other endpoints. So, I I don't think we should try to add it to the context for all endpoints.
We could create a new context inside the verify() endpoint and pass this new context with "code" to the verify service but it would be inconsistent with our existing controller/services. Currently our controllers takes care of the endpoint API and calls the corresponding services by passing all needed information directly through function arguments.

Reply:
oh i see, i missed the part that he was using `param` as in he's retrieving a param from the endpoint /{code}/